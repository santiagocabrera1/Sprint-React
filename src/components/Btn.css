.buttonComponent {
  display: flex;
  justify-content: center;
  align-items: center;
  width: max-content;
  height: 30px;
  padding: 0 14px;

  border: none;
  border-radius: 15px;
  background-color: var(--secondary-bg);

  color: var(--secondary-text-color);
  text-decoration: none;
  text-align: center;
  font-size: 12px;
}

.buttonComponent:hover {
  background-color: var(--color-green);
}
