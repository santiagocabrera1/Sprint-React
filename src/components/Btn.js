import React from 'react'
import { Link } from 'react-router-dom'

import './Btn.css'

const Btn = ({ to, handleClick, children }) => {
  return (
    <>
      {to ? (
        <Link to={to} className="buttonComponent">
          {children}
        </Link>
      ) : (
        <button className="buttonComponent" onClick={handleClick}>
          {children}
        </button>
      )}
    </>
  )
}

export default Btn
