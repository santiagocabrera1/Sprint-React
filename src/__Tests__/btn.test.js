import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Btn from '../components/Btn';
import { MemoryRouter } from 'react-router-dom';

// describe es para describir un conjunto de tests
describe('Btn: Componente que renderiza un boton o link con forma de Pill', () => {
  // cada test tiene una descripcion de lo que se va a testear
  // cada test tiene que ser lo mas pequeÃ±o y puntual posible
  test('renderiza un Link con una URL que se pasa por props', () => {
    // renderiza un component, al usar un Link es necesario que este dentro de un MemoryRouter
    render(<Btn to="/products">Products</Btn>, { wrapper: MemoryRouter })

    // buscar elemento por role link (hay uno solo en Btn)
    const btn = screen.getByRole('link')
    // espero que btn tenga un atributo href que sea igual al que pase por props como 'to'
    expect(btn).toHaveAttribute('href', '/products')
  })

  test('renderiza un Link con un contenido que se pasa por children', () => {
    render(<Btn to="/products">Products</Btn>, { wrapper: MemoryRouter })

    const btn = screen.getByRole('link')

    // espero que al renderizar btn matchee el snapshot que saco jest
    // un snapshot es una "foto" de un momento determinado, estan en __snapshoots__
    expect(btn).toMatchSnapshot()
  })

  test('renderiza un button con un contenido que se pasa por children', () => {
    render(<Btn>Eliminar Producto</Btn>)

    const btn = screen.getByRole('button')

    expect(btn).toMatchSnapshot()
    // query no falla si no encuentra el elemento, getByRole falla automaticamente
    expect(screen.queryByRole('link')).not.toBeInTheDocument()
  })
})
