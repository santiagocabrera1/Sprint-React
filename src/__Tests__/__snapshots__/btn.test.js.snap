// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn: Componente que renderiza un boton o link con forma de Pill renderiza un Link con un contenido que se pasa por children 1`] = `
<a
  class="buttonComponent"
  href="/products"
>
  Products
</a>
`;

exports[`Btn: Componente que renderiza un boton o link con forma de Pill renderiza un button con un contenido que se pasa por children 1`] = `
<button
  class="buttonComponent"
>
  Eliminar Producto
</button>
`;
